module/VuFind/src/VuFind/Controller/Plugin/CaptchaFactory.php

```diff
@@ -61,9 +63,22 @@ public function __invoke(ContainerInterface $container, $requestedName,
         if (!empty($options)) {
             throw new \Exception('Unexpected options sent to factory.');
         }
+
+        $config = $container->get(\VuFind\Config\PluginManager::class)
+            ->get('config');
+
+        $captchaTypes = isset($config->Captcha->types) ?
```

Comment:
Another opportunity for null coalescing (`??`).

Reply:
I also notice that the factories VuFind\Controller\Plugin\CaptchaFactory and VuFind\View\Helper\Root\CaptchaFactory appear to be identical. It might be worth considering a single reusable factory for these two situations (though naming it might be a challenge; I'll leave it to your discretion).